python-backend/app/api/mines/incidents/resources/mine_incidents.py

```diff
@@ -310,15 +310,16 @@ def put(self, mine_guid, mine_incident_guid):
 
     @classmethod
     def _handle_recommendations(cls, incident, recommendations):
-        for payload in recommendations:
-            edited_rec_guid = payload.get('mine_incident_recommendation_guid')
-            rec_string = payload.get('recommendation')
+        for recommendation in recommendations:
+            edited_rec_guid = recommendation.get('mine_incident_recommendation_guid')
+            rec_string = recommendation.get('recommendation')
             if rec_string is None or rec_string == '':
-                if payload.get('mine_incident_recommendation_guid') is not None:
+                if recommendation.get('mine_incident_recommendation_guid') is not None:
                     # Remove existing empty recommendations
                     rec = MineIncidentRecommendation.find_by_mine_incident_recommendation_guid(
                         edited_rec_guid)
-                    rec.soft_delete()
+                    rec.deleted_ind = True
+                    rec.save()
```

Comment:
Did this mean the delete wasn't being saved before?

Reply:
This is no change in behaviour. I just removed the `soft_delete` method, as per feedback on the Part 1 PR